src/main/java/org/itsallcode/openfasttrace/importer/asciidoc/AsciiDocImporter.java

```diff
@@ -1,26 +1,267 @@
+    private static final String ROLE_RATIONALE = ":rationale";
+    private static final String ROLE_SPECITEM = ":specitem";
+
+    private static final Logger LOG = Logger.getLogger(AsciiDocImporter.class.getName());
+
     private final InputFile file;
+    private final String content;
     private final ImportEventListener listener;
 
+    AsciiDocImporter(final String content, final ImportEventListener listener)
+    {
+        this.content = Objects.requireNonNull(content);
+        this.file = null;
+        this.listener = Objects.requireNonNull(listener);
+    }
+
     AsciiDocImporter(final InputFile file, final ImportEventListener listener)
     {
-        this.file = file;
-        this.listener = listener;
+        this.content = null;
+        this.file = Objects.requireNonNull(file);
+        this.listener = Objects.requireNonNull(listener);
+    }
+
+    private static List<String> getAttributeValueAsList(final StructuralNode block, final String attributeName)
+    {
+        return Optional.ofNullable(block.getAttribute(attributeName))
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .map(s -> StreamSupport.stream(Arrays.spliterator(s.split(",")), false)
+                        .map(String::trim)
+                        .toList())
+                .orElseGet(List::of);
+    }
+
+    private static List<String> getCoveredIds(final StructuralNode block)
+    {
+        return getAttributeValueAsList(block, ATTRIBUTE_OFT_COVERS);
+    }
+
+    private Location getLocation(final StructuralNode block)
+    {
+        return Location.builder()
+                .path(Optional.ofNullable(this.file).map(InputFile::getPath).orElse("verbatim"))
+                .line(block.getSourceLocation().getLineNumber()).build();
+    }
+
+    private Optional<String> getStringContent(final StructuralNode node)
+    {
+        switch (node.getContentModel())
+        {
+        case CONTENT_MODEL_SIMPLE:
+            LOG.fine("extracting SIMPLE content");
+            return Optional.ofNullable(node.getContent())
+                    .filter(String.class::isInstance)
+                    .map(String.class::cast);
+        default:
+            LOG.info(() -> "'%s' content model not (yet) supported [%s]".formatted(node.getContentModel(),
+                    getLocation(node)));
+            return Optional.empty();
+        }
+    }
+
+    // [impl->dsn~adoc-specification-item-title~1]
+    private void processSpecificationItemTitle(final StructuralNode block)
+    {
+        Optional.ofNullable(block.getAttribute(ATTRIBUTE_OFT_TITLE))
+                .filter(String.class::isInstance)
+                .map(String.class::cast)
+                .or(() -> Optional.ofNullable(block.getTitle()))
+                .ifPresent(s -> {
+                    LOG.fine(() -> String.format("setting spec item title: %s", s));
+                    this.listener.setTitle(s);
```

Comment:
```suggestion
                .ifPresent(title -> {
                    LOG.fine(() -> String.format("setting spec item title: %s", title));
                    this.listener.setTitle(title);
```
Is this log message really relevant?

Reply:
It was while debugging ;-) Thus the DEBUG level ...